editoast/src/views/authz.rs

```diff
@@ -46,22 +148,16 @@ enum AuthzError {
     Authorizer(#[from] AuthorizerError),
     #[error("Authorization error")]
     Authz(#[from] AuthorizationError),
+    #[error("Bad grants operation : {message}")]
+    BadGrantsOperation { message: String },
 }
 
 #[derive(Debug, thiserror::Error, EditoastError)]
 #[editoast_error(base_id = "authz")]
-enum NoSuchUserError {
-    #[error("No user with ID {user_id} found")]
+enum ResourceError {
+    #[error("Resource type {resource_type} not found")]
     #[editoast_error(status = 404)]
-    NoSuchUser { user_id: i64 },
-}
-
-#[derive(Debug, serde::Serialize, utoipa::ToSchema)]
-#[cfg_attr(test, derive(serde::Deserialize, PartialEq))]
-struct WhoamiResponse {
-    id: i64,
-    name: String,
-    roles: Vec<Role>,
+    ResourceTypeNotFound { resource_type: String },
```

Comment:
```suggestion
    UnknownResourceType { resource_type: String },
```

nit++: `-NotFound` strikes me as a an error for data that may or may not be there, whereas here we know for sure the resource type in question doesn't exist